static/js/board_carousel.js

```diff
@@ -149,33 +152,54 @@ function deleteElement(bobid){
   if ($('#slideshow').hasClass('initialized')) {
     $('#slideshow').removeClass('initialized');
   }
-
   // reinit the carousel
   $('#slideshow').carousel({
-    fullWidth: true
+    fullWidth: true,
+    onCycleTo: function(activeItem){
+      updateVoteLabel(activeItem);
+      loadVideo(activeItem);
+    }
   });
   // force move to next slide
   carouselControl("right");
 }
 
 /**
-  * Updates the label with the votes label for currently active bob
+  * @param {object} activeItem - Vanilla JS Object of activeItem
+  * Updates the label with the votes label for currently activeItem
 */
-function updateVoteLabel(){
+function updateVoteLabel(activeItem){
   var $labelToUpdate = $("#votes");
-  var activeBobID = $("#slideshow").find(".active").attr("id");
-  var votes =  $.get('/api/bobs/' + activeBobID + "/votes", function(res){
+  var $activeBobID = ($(activeItem).attr("id"));
+  var votes =  $.get('/api/bobs/' + $activeBobID + "/votes", function(res){
     $labelToUpdate.attr("data-badge-caption", "+" + res.votes);
   });
 }
 
 /**
   * Updates the label with the votes returned from socket
-  * parpm {object} res - response from the socekt which contains bobid and votes
+  * @param {object} res - response from the socekt which contains bobid and votes
```

Comment:
socket
Always on the lookout for typos
![](http://apollo-na-uploads.s3.amazonaws.com/1439790276/batsignal.png)